bot.py

```diff
@@ -193,22 +193,27 @@ def check(m):
 			"guess"
 		)
 
-	message = await bot.wait_for("message", check=check)
-	guess = message.content.split()[-1]
+	while True:
+		message = await bot.wait_for("message", check=check)
+		guess = message.content.split()[-1]
+		profile = await userprofile.load_profile(message.author.id)
```

Comment:
Again, you are still hitting the database every time you receive a message.

Reply:
See the commit I've pushed to this branch

Reply:
please see my response to your prior comment [here](https://github.com/dyc3/discord-pokemon-battles/pull/97#discussion_r617788028). i understand what you're doing and why its not smart to access the database excessively, but the user experience you're suggesting seems a little odd to me. why should someone who doesn't have a profile be allowed to play the game at all? if i were a user i would want to know that i won't get my pokemon even if i guess correctly before i play the game, not after